cast/src/starknet_commands/call.rs

```diff
@@ -44,7 +48,16 @@ pub async fn call(
             })
             .collect::<Result<Vec<_>>>()?,
     };
-    let res = provider.call(function_call, block_id).await?;
+    let res = provider.call(function_call, block_id).await;
 
-    Ok(res)
+    match res {
+        Ok(res) => Ok(res),
+        Err(error) => match error {
+            Other(RpcError(Code(error))) | StarknetError(error) => {
+                Err(anyhow!(get_rpc_error_message(error)))
+            }
+            Other(RpcError(Unknown(error))) => Err(anyhow!(error.message)),
+            _ => Err(anyhow!("Other RPC error")),
```

Comment:
Are there any codes not declared in our codebase, that were added to starknet-spec? 
Maybe we could change message to something like `"Unknown RPC error {}", code` ?